### DISCUSSION TITLE
/DS3

### DISCUSSION THREAD

**Discussion by ibm5110** [keirf/greaseweazle#289] [Ideas]
The Shugart bus supports four drive selects, not only three. According to the schematic of the V4 /DS3 is labelled as SPARE2 and connected to PA7. Can the firmware and tools be made to support this drive select line, e.g. with the option "--drive 3" ?

**Comment by keirf** [comment]
It certainly could, though is anyone actually connecting four drives to their Greaseweazle?

The other reason to question this is that some users like to sue the spare pins for driving other signals. Eject, REDWC, etc etc. Currently we have 3 user pins 2,4,6. One is often density select, and if another becomes a *fourth* drive select, we only have one truly spare pin. Which isn't many.

Arguably though, more user pins could be made available on a separate header on a future board revision.

So I suppose TLDR: Do you actually want to use four drives, or is this merely an observation and a nice to have?

**Reply by ibm5110** [reply]
Actually, I'm struggling with the four drive selects restriction of the Shugart bus ;-)
My setup consists of four drives: 1x 5¼" 48tpi, 1x 5¼" 96tpi (DD/HD), 1x 3½" 135tpi, 1x 8"
The 8" drive is currently in a separate box with a separate data cable.
I would want to attach a 3" drive and a 100tpi 5¼" drive, too, but this another story.

Right now I have to change the cables when I want to change from/to 8". Since it should be a static setup for data transfers, it would be nice to cable the drives only once.

A workaround could be to have to issue a "gw set pin 6 L" beforehand, but this would only work _if_ I can tell the software/firmware to not issue and wanting a drive select.

Another workaround could be to use one of the user pins to switch between two Shugart busses. This could be done with a set pin command. But this involves a daughter board with a multiplexer for the drive select/motor on lines.

BTW Eject is not really needed, just use a drive without motor eject. RedWC is only needed for very old 8" drives and only when writing, so is also of very small interest. I think using four drives is much more common ;-)

**Comment by keirf** [comment]
Have you considered two Greasweazles? I don't know what your host is, but on Linux you can use udev rules to name the Weazles under `/dev` and then address the specific Weazle device using `--device` option to `gw`. I suppose on Windows this doesn't work though (you would need to know the `COMn` numbers and these can change). But it would let you statically cable all your drives. You seem like someone who might use Linux/UNIX though haha.

**Reply by keirf** [reply]
Another option would be to add `--serial` to the `gw` options list, so that a Greasweazle can be identified by full serial. It's a bit verbose maybe, unless you can hide it behind a script or batch file perhaps.
